### DISCUSSION TITLE
ReValidate fields after field error not working/weird

### DISCUSSION THREAD

**Discussion by mbretter** [react-hook-form/react-hook-form#5552] [Q&A] [answered]
**Describe the bug**
Field errors are not reset after input field has changed, at least before first submit

**To Reproduce**

1. Go to the codesandbox
2. Do not enter any value
3. Click on "SetErrors"
4. Enter a value into the Username field
5. Field error should disappear, but it does not
6. Click the submit button
7. Click "Seterrors" again
8. Enter a username
9. Error disappears, as expected

**Codesandbox link (Required)**
https://codesandbox.io/s/rhf-basic-forked-wpgk0?file=/src/App.js

**Expected behavior**
The field errors should disappear after changing the input fields value.

**Desktop (please complete the following information):**

- OS: Linux
- Browser [chrome 87, firefox 89]

**Comment by bluebill1049** [comment, marked as answer]
> An error that is not associated with an input field will be persisted until cleared with `clearErrors`.

<img width="949" alt="Screen Shot 2021-06-10 at 4 14 37 pm" src="https://user-images.githubusercontent.com/10513364/121474116-f5f4ae80-ca06-11eb-8d1e-09e6be471200.png">

**Reply by mbretter** [reply]
Why the error is then cleared after the form has been submitted the first time?

**Reply by mbretter** [reply]
And why the error should not be associated with an input field in this example?
Controllers field name = "name", setError is invoked using "name", error and field are definitely connected together, or am I missing something?

**Reply by bluebill1049** [reply]
ok, i miss look the input `name` is the same...

your controller doesn't have validation, that's why it passes the validation during submit. what's in the rules is going to tested during submission, and its expected behavior.
